billing/actions/accounts.py

```diff
@@ -329,3 +324,15 @@ def toggle_delinquent_status(account_ids: List[int]):
         When(delinquent=True, then=Value(False)),
         default=Value(True)
     ))
+
+
+def charge_pending_invoices(account_id: str):
+    account = Account.objects.get(id=account_id)
+    pending_invoices = account.invoices.payable().only('pk')
+    logger.info('charge-pending-invoices', pending_invoices=pending_invoices)
+    payments = [
+        invoices.pay_with_account_credit_cards(invoice.pk)
+        for invoice in pending_invoices
+    ]
+    if len(payments) == len(pending_invoices):
```

Comment:
payments is a list of Maybe[Transaction], I believe its length will always we the length of pending_invoices.
You should filter our the Nones if you want to compare.

Reply:
Now I'm filtering with this: 
`payment_transaction and payment_transaction.success`

Reply:
you're right I missed that test, but you still need to also remove the Nones
(look at the code of pay_with_account_credit_cards)

Reply:
I reckon that filtering with this
```
if payment_transaction:
   payment_transactions.append(payment_transaction)
```
it's enough. `pay_with_account_credit_cards` just return the transaction if it went well. However now I'm catching precondition errors in order to return this errors.

Reply:
also that: you''re doing multiple operations in a loop and each can throw a precondition error, so in order to correctly build the status of _all_ the operations you can't just let exceptions bubble out. Right now in the case of 3 invoices if two get paid but the last one fails you get a blunt 500 error, whereas you should most likely say: 2 were paid, one failed.

Reply:
about the first remark, please look at the _types_ : https://github.com/skioo/django-customer-billing/blob/master/billing/actions/invoices.py#L16
That tells you that the function may return nulls.

The comment (And of course the body of the function) give even more information: https://github.com/skioo/django-customer-billing/blob/master/billing/actions/invoices.py#L23

Reply:
As we discuss I have simplified `update_accounts_delinquent_status_command`. Now it doesn't need any threshold param. 
Now action function `compute_account_violations` looks like this:
```
def account_violates_delinquent_criteria(account: Account) -> bool:
    return account.invoices.filter(status=Invoice.PENDING).count() > 0
```
So I can use it after charge pending invoices at `charge_pending_invoices` action in this way:
```
if not account_violates_delinquent_criteria(account):
     account.mark_as_compliant()
```
The idea is that each new criteria we have to add to know if an account is delinquent has to be encapsulated inside this function without params (`compute_account_violations`). Maybe in the future we will need to add settings variables in order to define more complex thresholds.

What do you think now? Do you think naming is correct?